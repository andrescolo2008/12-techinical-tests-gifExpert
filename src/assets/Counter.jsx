import { useState } from "react"

import PropTypes from "prop-types"

export const Counter = ({contador}) => {
    const [counter, setCounter] = useState(contador)
    const [counterValue, setCounterValue] = useState(2)

    const sumar= () =>{
       const  count=counter+counterValue;
       setCounter(count)
      
       
    }

    const restar= ( ) =>{
        if(counter===0) return
        const  count=counter-counterValue;
        setCounter(count)  
        
     }

     const handleChange = ({target}) => {
        let value = target.value;

        // Si el valor no es un string vacío, convierte el valor a un número entero
        // de lo contrario, deja que el valor del input sea el mismo (string vacío)
        if(value !== ''){
            value=parseInt(target.value, 10)
        }
       
      
        // Actualiza el estado counterValue con el nuevo valor
        setCounterValue(value);
       
       
      };


      const evento= ( ) =>{
            console.log(counterValue);
            
    }
    return (
    <>
    <h1>Counter</h1>
    <h2>{counter}</h2>
    <input type="text"
    value={counterValue}
    onChange={handleChange}
     />
    <button onClick={sumar}>sumar</button>
    <button onClick={restar}>restar</button>
    <button onClick={evento}>eventor</button>
    
    </>
    )
}

Counter.propTypes={
    contador:PropTypes.number,
    base:PropTypes.number.isRequired

}

Counter.defaultProps ={
    contador:1,
    base:2
}
