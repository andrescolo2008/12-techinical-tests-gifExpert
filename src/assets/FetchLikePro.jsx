import { useEffect, useState } from "react";


export const FetchLikePro = () => {
    // const fetchApi=fetchData('http://api.giphy.com/v1/gifs/search?api_key=tOuGo3jNBa9ARQ5eyCuR6crElbBVtfnJ&q&limit=5')
    // const [data, setdata] = useState(null)
    // http://api.giphy.com/v1/gifs/search?api_key=tOuGo3jNBa9ARQ5eyCuR6crElbBVtfnJ&q&limit=5
    const [image, setimage] = useState('')
    
    const fetchGif = async () => {
        try {
            const url = 'https://api.giphy.com/v1/gifs/random?apiKey=bu7WEF9trqzmxwWQygW1Mh70nRKi3ehg';
            const resp = await fetch(url);
            const { data } = await resp.json();
            const imageUrl = data.images.downsized_medium.url;
            setimage(imageUrl);
        } catch (error) {
            console.error('Error fetching GIF:', error);
        }
    };

     useEffect(() => {
            
            
            fetchGif();
        }, []);//Llamar a fetchGif solo una vez al montar el componente
        
    
        const handeClick = ( ) =>{
        fetchGif()
        }
  return (
    
  <div>
    <div>FetchLikePro</div>
    <div className="card">
        <ul>
            {
            <img src={image} alt="gif" />
            }
        </ul>
        <button onClick={handeClick}>siguiente gif</button>
    </div>

    </div>
  )
}
