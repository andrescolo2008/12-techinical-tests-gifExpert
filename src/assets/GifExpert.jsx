import React, { useState } from 'react'
import { AddCategory, Gifgrid } from './components'
import { number } from 'prop-types'

// import {AddCategory} from './components/AddCategory'
// import { Gifgrid } from './components/Gifgrid'



export const GifExpert = () => {
  
const [categories, setCategories] = useState(['one punch man',])
const [number, setNumber] = useState(0)

const onAddCategory= ( newCategory) =>{
        
  // console.log(newCategory);
  if(categories.includes(newCategory)) return;
  
  // setCategories([...categories,newCategory])
  setCategories([newCategory])
}

const onCategorySubmit = (newCategory) => {
  onAddCategory(newCategory);
}

const onNewNumber = (newNumber) =>{
setNumber(newNumber)



}

return (
    
      <>
      <div>GifExpert</div>


      <AddCategory  onCategorySubmita={onCategorySubmit} onNewNumber={onNewNumber}  />

        {/* listado de  gifs */}
          
          {
          categories.map(category =>(
            < Gifgrid category={category} key={category} number={number} />)) 
          
          }


      </>


  )

}
