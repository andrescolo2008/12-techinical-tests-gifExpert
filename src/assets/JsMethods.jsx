import React from 'react'

export const JsMethods = () => {
  //   const url ='www/andres/com'
  //   const array= (data ) =>{
  //       return data.split("/").slice(1,2)
  //   }

  //  console.log(array(url));

  //función para sacar números pares e impares de una función

   
   const numeros2=[34,56,12,2,4,5,78,1000,3]
   const texto="This document will change on an as needed basis"

  const pares= ( ) =>{
    const par= numeros.filter(numero=>numero %2 ===0)
    return par;
    
  }
  

  const impares= ( ) =>{
    const par= numeros.filter(numero=>numero %2 !== 0)
    return par;
    
  }
  
  const ordenamiento= ( ) =>{
    const ord =numeros2.sort((a,b)=>b-a)
    return ord
  }
    
  const separarcion= ( ) =>{
    const sep= texto.split(" ")
    return sep
  }

  const numeros=[1,2,3,5,6,7,8,10,12,14,22]
  const letras=["a","b","c"]

  const mezcla= (element) =>{
    const newArray = numeros.map((numero, index) => {
      return numero.toString() + letras[index % letras.length];
    });
     return newArray;
  }

  const personas= [
    {name:'andres',age:34},
    {name:'lucas',age:26},
    {name:'lTomás',age:45},
    {name:'Luna',age:13},

  ]

 personas.some((persona) =>{
    return persona.age >16
  })

  const personasfiltradas=personas.filter((persona) =>persona.age >  16)
  
  const muestraPersonasFiltradas= personasfiltradas.map(   
    (persona,index)=>(  <div key={index}> {persona.age} - {persona.name}</div>)
  )
  console.log(personasfiltradas);
  


  return (
    <>
    <div>Hola , Aquí practicaré algunos médtodos de js</div>
    {/* <div>{pares(numeros).join(', ')}</div>
    <div>{impares(numeros).join(', ')}</div>
    <div>{ordenamiento(numeros2).join('#,# ')}</div>
    <div>{separarcion(texto).join(',')}</div>
    <div>{mezcla()}</div> */}
   <div>resultado de personas filtradas</div>
   {/* 
   
    const muestraPersonasFiltradas= personasfiltradas.map(   
    (persona,index)=>(  <div key={index}> {persona.age} - {persona.name}</div>)
  )
   = a 

   {personasFiltradas.map((persona, index) => (
        <div key={index}>{persona.name} - {persona.age}</div>
      ))} */}
   {/* <div>{muestraPersonasFiltradas}</div> */}

   <ol>
   {
   personasfiltradas.map( (persona,numeroPersona)=>
     
   { 
     return(  
         
         <li key={numeroPersona}> {persona.age} - {persona.name}</li>
         );
        }
  
   
    )
   }
        </ol>
    </>
  )
}

