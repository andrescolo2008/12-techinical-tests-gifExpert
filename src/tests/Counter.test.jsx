
import React from "react";

import { Counter } from "../assets/Counter";

import { render } from "@testing-library/react";

describe('pruebas en < Counter.jsx>',()=>{

test('deb de mostrar el valor del counter', () => { 
    // const variable='amigos'    
    // const message='hola amigos';
    // const message2=`hola ${variable}`;

    // expect(message2).toBe(message)
    const base =1

    const {container}= render(
        < Counter base={base}/>

        )
        expect(container).toMatchSnapshot()


   })
})

