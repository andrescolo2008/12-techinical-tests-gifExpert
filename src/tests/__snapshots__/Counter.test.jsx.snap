// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en < Counter.jsx> deb de mostrar el valor del counter 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <h2>
    1
  </h2>
  <input
    type="text"
    value="2"
  />
  <button>
    sumar
  </button>
  <button>
    restar
  </button>
  <button>
    eventor
  </button>
</div>
`;
